test/EvalTests.hs

```diff
@@ -151,19 +151,34 @@ testNegativeBoardAccess = TestCase (
 --   This is reflected as the output Vs "X"
 evalTicTacToe :: IO (Bool, String)
 evalTicTacToe = do
-   res <- parseGameFile $ examplesPath ++ "TicTacToe.bgl"
+   [(b, _, r)] <- evalFile (examplesPath ++ "TicTacToe.bgl") [("result", Vs "X")] buf
+   return (b, r)
+   where
+      buf        = (moves, [], 0)
+      moves      = map Vt (map (\_p -> [Vi (fst _p), Vi (snd _p)]) coords)
+      coords     = [(1, 1), (2, 1), (2, 2), (3, 1), (3, 3)]
+
+-- | Evaluates many different expressions that contain while loops
+evalWhile :: IO [(Bool, String, String)]
+evalWhile = evalFile (examplesPath ++ "While.bgl") vs ([], [], 0)
+   where
+      vs = [("false", Vb False), ("ten1", Vi 10), ("ten2", Vi 10), ("ten3", Vi 10),
+            ("tenOne", Vt [Vi 10, Vi 1]), ("ten4", Vi 10), ("thirty", Vi 30), ("twenty", Vi 20),
+            ("eleven", Vi 11), ("five", Vi 5), ("fifteen", Vi 15), ("twentyNine", Vi 29)]
```

Comment:
Awesome! I was going to ask about testing those claims made above, but this is it 👍 .